### ISSUE TITLE
Plugins no longer load when using Node 22.12.0 and 23.*

### ISSUE THREAD

**Issue by xt0rted** [tailwindlabs/tailwindcss#15374]
**What version of Tailwind CSS are you using?**

v3.4.16

**What build tool (or framework if it abstracts the build tool) are you using?**

tailwind cli

**What version of Node.js are you using?**

v22.12.0
v23.*

**What browser are you using?**

N/A

**What operating system are you using?**

Windows

**Reproduction URL**

N/A

**Describe your issue**

It seems module resolution changed in Node [22.12.0](https://github.com/nodejs/node/releases/tag/v22.12.0) (as well as v23) and now tailwind no longer loads plugins. The only one I'm using is `@tailwindcss/forms` and with Node v22.11.0 it's included in the output, but with v22.12.0 or v23.* it's not.

My `package.json` is set to `type": "module"` and my config is named `tailwind.config.js`.

I can resolve this issue two different ways:

1. Rename my config to use `.mjs`

2. Pass the `--no-experimental-require-module` flag when calling tailwind like so

    ```console
    NODE_OPTIONS=\"--no-experimental-require-module\" tailwindcss -i ./Styles/base.css --postcss -o ./wwwroot/css/site.css
    ```

ref: https://github.com/nodejs/node/pull/55085
ref: https://github.com/nodejs/node/issues/56155

**Comment by RobinMalfait** [comment]
Hey!

Can you share a minimal reproduction repo so we can take a look?

**Comment by xt0rted** [comment]
Here you go https://github.com/xt0rted-test/tailwind-plugins

While setting that up I found a few more things:

1. The bug requires the config file to import `@tailwindcss/forms` and `tailwindcss/defaultTheme`. In my testing yesterday it seems to only happen with the forms plugin so it may vary some based on what's in your config.
2. Adding a file extension to `tailwindcss/defaultTheme` looks to be a third way to resolve the issue.

**Comment by zachczx** [comment]
I can confirm node 22.12 and node 23.4 broke daisyui plugin. Reverting to node 22.11 solved it.

**Comment by xt0rted** [comment]
This is where I was pointed to when I commented on the original PR that changed this flag https://github.com/nodejs/node/issues/56155#issuecomment-2525032281

**Comment by thecrypticace** [comment]
I've tracked this down to an issue potentially with Jiti.

The issue happens when a `require(esm)` call _fails_ and then Jiti is subsequently used to load that file.

reproduction here: https://github.com/thecrypticace/node-22-jiti-bug

I'm going to file a bug with Jiti and also see if I can track this within Jiti itself

**Comment by thecrypticace** [comment]
Because using Jiti alone doesn't appear to be broken I'd suggest renaming your config file to `tailwind.config.mjs` for the time being since we force the use of Jiti in that case.

**Comment by thecrypticace** [comment]
I published v3.4.17 which always runs Jiti for configs. Hopefully this will work around the Node v22.12+ issues.

@xt0rted @zachczx can y'all see if things are working again for you?

**Comment by xt0rted** [comment]
@thecrypticace v3.4.17 seems to be working with node 22.12.0 now, but I'll have to do a more in-depth test later to fully confirm.

**Comment by onlime** [comment]
Thanks @thecrypticace for v3.4.17, confirm it working as well in my project, renamed `tailwind.config.mjs` back to `tailwind.config.js` / v22.12.0

**Comment by onlime** [comment]
It should now be hotfixed in jiti:

https://github.com/unjs/jiti/issues/346#issuecomment-2549665679

> [v1.21.7](https://github.com/unjs/jiti/releases/tag/v1.21.7) + [v2.4.2](https://github.com/unjs/jiti/releases/tag/v2.4.2) released with hotfix. (confirmed fix on your reproduction)

**Comment by xt0rted** [comment]
@thecrypticace after more testing this looks to have fixed my issue

**Comment by zachczx** [comment]
> I published v3.4.17 which always runs Jiti for configs. Hopefully this will work around the Node v22.12+ issues.
> 
> @xt0rted @zachczx can y'all see if things are working again for you?

All good on my end too, thank you!!
